src/taipy/core/data/_data_manager.py

```diff
@@ -64,13 +65,15 @@ def __create(
         cls, data_node_config: DataNodeConfig, owner_id: Optional[str], parent_ids: Optional[Set[str]]
     ) -> DataNode:
         try:
+            version = Config.global_config.version or "latest"
```

Comment:
I would create a dedicated service/class to retrieve the correct version. The purpose is to have the whole versioning logic in a single location.

```suggestion
            version = Version.get_version()
```